spring-boot-modules/spring-boot-keycloak/spring-boot-mvc-client/src/main/java/com/baeldung/boot/keycloak/client/SecurityConfig.java

```diff
@@ -0,0 +1,76 @@
+package com.baeldung.boot.keycloak.client;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.Customizer;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper;
+import org.springframework.security.oauth2.client.oidc.web.logout.OidcClientInitiatedLogoutSuccessHandler;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.core.oidc.user.OidcUserAuthority;
+import org.springframework.security.web.SecurityFilterChain;
+
+@Configuration
+@EnableWebSecurity
+@EnableMethodSecurity
+public class SecurityConfig {
+
+  static interface AuthoritiesConverter
+      extends Converter<Map<String, Object>, Collection<GrantedAuthority>> {
+  }
+
+  @Bean
+  AuthoritiesConverter realmRolesAuthoritiesConverter() {
+    return claims -> {
+      final var realmAccess = Optional.ofNullable((Map<String, Object>) claims.get("realm_access"));
+      final var roles =
```

Comment:
These `final` are unnecessary. It's a short method and everything is effectively final.

Reply:
The method being short does not make a local variable `final`.

These references are not intended to be changed. The `final` keyword declares that.